### PR TITLE
Pass TestSearch benchmark consistently (Add browse_website TOKENS_TO_TRIGGER_SUMMARY)

### PR DESCRIPTION
**Pull request by lc0rp**
### Background
TestSearch was failing half the time because of browse_website summarization.

Also, for shorter text, this PR saves us an extra LLM call, which is faster and cheaper, without degrading performance - the main LLM can pull answers from the text, just like the summarizing LLM would have.

#### Benchmarks:
1. With this fix, if --nc is used (no timeout), Auto-GPT passes all agbenchmarks (version from pypi) except r2.1, r2.2, r3 (answers are correct but precision is tricky) and d4 (it starts the web server but cannot terminate the process)

### Changes
- Added `TOKENS_TO_TRIGGER_SUMMARY = 50,` 
- Altered `browse_website` to only trigger summarization if the text is longer than `TOKENS_TO_TRIGGER_SUMMARY`.
- Fixed minor typing issues found by mypy

### Documentation
N/A

### Test Plan
- Ran existing tests
- Tested locally
- Passed benchmarks

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes.
- [X] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

File: autogpt/commands/web_selenium.py [hunk 1]
```diff
@@ -66,23 +68,31 @@ def browse_website(url: str, question: str, agent: Agent) -> str:
...
+    else:
+        add_header(driver)
 
-    add_header(driver)
-    summary = summarize_memorize_webpage(url, text, question, agent, driver)
-    links = scrape_links_with_selenium(driver, url)
+        token_length = count_string_tokens(text, agent.llm.name)
+        if token_length > TOKENS_TO_TRIGGER_SUMMARY:
+            summary = summarize_memorize_webpage(url, text, question, agent, driver)
+            links = scrape_links_with_selenium(driver, url)
```
Comment (collijk) [inline autogpt/commands/web_selenium.py]: I get not summarizing short webpages, but why do we only scrape links if there are more than 50 tokens?

File: autogpt/commands/web_selenium.py [hunk 2]
```diff
@@ -66,23 +68,31 @@ def browse_website(url: str, question: str, agent: Agent) -> str:
...
 
-    # Limit links to 5
-    if len(links) > 5:
-        links = links[:5]
-    close_browser(driver)
-    return f"Answer gathered from website: {summary}\n\nLinks: {links}"
+            # Limit links to 5
+            if len(links) > 5:
```
Comment (collijk) [inline autogpt/commands/web_selenium.py]: Can you lift the max links to a constant as well?

File: autogpt/commands/web_selenium.py (hunk 1 above)
Comment (lc0rp) [reply autogpt/commands/web_selenium.py]: Good point. Updating.

File: autogpt/commands/web_selenium.py (hunk 2 above)
Comment (lc0rp) [reply autogpt/commands/web_selenium.py]: Yeah - I wonder whether we should also randomize the list? That could reduce loops.

Comment (collijk) [reply autogpt/commands/web_selenium.py]: I don't think you want to randomize.  The link order contains information.

Comment (lc0rp) [reply autogpt/commands/web_selenium.py]: Updated.

**Review by collijk — APPROVED** [review]

**Pull request merged**